Mathlib/Data/List/Basic.lean

```diff
@@ -2862,6 +2862,13 @@ theorem length_lookmap (l : List α) : length (l.lookmap f) = length l := by
 end Lookmap
 
 /-! ### filter -/
+
+theorem length_eq_length_filter_add {l : List (α)} (f : α → Bool) :
+    l.length = (l.filter f).length + (l.filter (Bool.not ∘ f)).length := by
```

Comment:
Should this be phrased with `(! ·)` instead of `Bool.not`?

Proof could probably also be
```
theorem length_eq_length_filter_add {l : List (α)} (f : α → Bool) :
    l.length = (l.filter f).length + (l.filter ((! ·) ∘ f)).length := by
  simp_rw [← List.countP_eq_length_filter, List.length_eq_countP_add_countP f, Function.comp_def, Bool.not_eq_true, Bool.decide_eq_false]
```
(the Bool/Prop confusion is annoying...)

Reply:
\comp reminds me of point-free style, but this is a golf, so sure. Even better seems to be to use pointful notation and remove the comp - it seems to golf the proof a bit and doesn't mess things up down the road.